packages/smithy-core/tests/functional/test_retries.py

```diff
@@ -0,0 +1,133 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from asyncio import gather, sleep
+
+import pytest
+from smithy_core.exceptions import CallError, RetryError
+from smithy_core.interfaces import retries as retries_interface
+from smithy_core.retries import (
+    ExponentialBackoffJitterType,
+    ExponentialRetryBackoffStrategy,
+    StandardRetryQuota,
+    StandardRetryStrategy,
+)
+
+
+async def retry_operation(
```

Comment:
We should probably add a TODO to refactor this to use a generated client from `smithy-testing` once we have that finalized. This is a great first step to make sure we have coverage in the mean time though.